dashboard/app/models/census/other_curriculum_offering.rb

```diff
@@ -17,4 +17,102 @@
 #
 
 class Census::OtherCurriculumOffering < ApplicationRecord
+  belongs_to :school, required: true
+
+  validates_presence_of :curriculum_provider_name, :course
+  validates :school_year, presence: true, numericality: {greater_than_or_equal_to: 2015, less_than_or_equal_to: 2030}
+
+  SUPPORTED_PROVIDERS = %w(
+    TEALS
+  ).freeze
+
+  TEALS_COURSE_CODES = [
+    'Intro Full',
+    'Intro S2',
+    'Other',
+    'AP',
+    'AP CSP',
+    'Intro Both',
+    'Intro S1',
+    'Post AP',
+    'Python S2'
+  ].freeze
+
+  def self.school_id(provider_code, row_hash)
+    case provider_code
+    when 'TEALS'
+      row_hash['School ID']
+    else
+      raise ArgumentError.new("#{provider_code} is not supported.")
+    end
+  end
+
+  def self.get_courses(provider_code, row_hash)
+    case provider_code
+    when 'TEALS'
+      TEALS_COURSE_CODES.select {|course| course == row_hash['Course']}
+    else
+      raise ArgumentError.new("#{provider_code} is not supported.")
+    end
+  end
+
+  def self.seed_from_csv(provider_code, school_year, filename)
+    ActiveRecord::Base.transaction do
+      CSV.foreach(filename, {headers: true}) do |row|
+        row_hash = row.to_hash
+        # Remove leading zero because we imported NCES IDs into School.id without the leading zero.
+        school_id = school_id(provider_code, row_hash)&.sub!(/^0/, "")
```

Comment:
@mirlew The TEALS data specifies the actual NCES ID, but we imported NCES IDs with out leading zeros (tech debt!) into our Schools table.  This logic assumes the school id in the CSV is an NCES ID and strips the first leading zero (if it exists) so that it can be matched with the an existing School.  Any school with a non-NCES ID in this CSV will be skipped due to no match found, particularly if the non-NCES ID has a leading zero.

Reply:
Do you know how many we're losing vs. getting?

Reply:
Also, is there a chance we can reimport after we update our NCES list to get schools we couldn't get at this time?

Reply:
387 of the 407 school lookups are succeeding.  A warning is written to the log if a lookup fails:
`Other Curriculum Offering seeding: skipping unknown school with NCES ID: 550007402637`